### ISSUE TITLE
luarocks module missing

### ISSUE THREAD

**Issue by MrChrisJohnson** [torch/nngraph#52]
The documentation referenced at https://www.cs.ox.ac.uk/people/nando.defreitas/machinelearning/practicals/practical5.pdf suggests installing nngraph using LuaRocks

```
$ luarocks install nngraph
```
However, when I run this I get

```
[~]$ luarocks install nngraph
Warning: falling back to curl - install luasec to get native HTTPS support

Error: No results matching query were found.
```

Additionally, if you try searching for the package at https://luarocks.org/ it's not found.

**Comment by soumith** [comment]
hey Chris, it's assumed that you install torch from https://torch.ch and use the luarocks that's shipped via torch.

We maintain our own manifest that additionally holds quite a few packages that are not part of the luarocks.org manifest.
Our rocks are located here: https://github.com/torch/rocks

**Comment by MrChrisJohnson** [comment]
I must have installed LuaRocks before installing Torch. Any tips on how to remedy? I'm trying to install on OSX btw.

**Comment by soumith** [comment]
put ~/torch/install/bin in your $PATH as the first entry. That way the luarocks installed via torch get's precedence.

**Comment by MrChrisJohnson** [comment]
Great, that fixed it, thanks! Closing this issue.

**Comment by cpdean** [comment]
if you're still having this issue, be sure that you're only using one "Rocks Tree".  If you still can't find `nngraph`, run `luarocks` to see the currently used trees:

example below:
```
$ luarocks install nngraph

Error: No results matching query were found.
$ luarocks

LuaRocks 2.2.0beta1, a module deployment system for Lua

NAME
        /Users/conrad/torch/install/bin/luarocks - LuaRocks main command-line interface

SYNOPSIS
        /Users/conrad/torch/install/bin/luarocks [--from=<server> | --only-from=<server>] [--to=<tree>] [VAR=VALUE]... <command> [<argument>]

GENERAL OPTIONS
        These apply to all commands, as appropriate:

        --server=<server>      Fetch rocks/rockspecs from this server
                               (takes priority over config file)
        --only-server=<server> Fetch rocks/rockspecs from this server only
                               (overrides any entries in the config file)
        --only-sources=<url>   Restrict downloads to paths matching the
                               given URL.
        --tree=<tree>          Which tree to operate on.
        --local                Use the tree in the user's home directory.
                               To enable it, see '/Users/conrad/torch/install/bin/luarocks help path'.
        --verbose              Display verbose output of commands executed.
        --timeout=<seconds>    Timeout on network operations, in seconds.
                               0 means no timeout (wait forever).
                               Default is 30.

VARIABLES
        Variables from the "variables" table of the configuration file
        can be overriden with VAR=VALUE assignments.

COMMANDS
        build
                Build/compile a rock.
        doc
                Shows documentation for an installed rock.
        download
                Download a specific rock file from a rocks server.
        help
                Help on commands. Type '/Users/conrad/torch/install/bin/luarocks help <command>' for more.
        install
                Install a rock.
        lint
                Check syntax of a rockspec.
        list
                Lists currently installed rocks.
        make
                Compile package in current directory using a rockspec.
        new_version
                Auto-write a rockspec for a new version of a rock.
        pack
                Create a rock, packing sources or binaries.
        path
                Return the currently configured package path.
        purge
                Remove all installed rocks from a tree.
        remove
                Uninstall a rock.
        search
                Query the LuaRocks servers.
        show
                Shows information about an installed rock.
        unpack
                Unpack the contents of a rock.
        upload
                Upload a rockspec to the public rocks repository.
        write_rockspec
                Write a template for a rockspec file.

CONFIGURATION
        Lua version: 5.1
        Configuration files:
                System: /Users/conrad/torch/install/etc/luarocks/config.lua (ok)
                User  : /Users/conrad/.luarocks/config.lua (ok)

        Rocks trees in use:
                /Users/conrad/.luarocks
                /Users/conrad/torch/install
$ mv ~/.luarocks/config.lua{,.backup}
$ luarocks install nngraph
Installing https://raw.githubusercontent.com/torch/rocks/master/nngraph-scm-1.rockspec...
Using https://raw.githubusercontent.com/torch/rocks/master/nngraph-scm-1.rockspec... switching to 'build' mode
...
```

**Comment by Quasimondo** [comment]
I'm having the same problem and none of the above fixes worked for me. But I eventually managed to get nngraph installed using this command:

`luarocks --from=https://raw.githubusercontent.com/torch/rocks/master/ install nngraph`

**Comment by hughperkins** [comment]
I added the following to the list of things I do prior to using luarocks etc:

```
eval $(luarocks path --bin)
```

**Comment by diggerdu** [comment]
thanks

**Comment by delip** [comment]
The only solution works for me is this in my .bashrc :smile: 

`alias luarocks='sudo luarocks --from=https://raw.githubusercontent.com/torch/rocks/master/'
`

**Comment by soumith** [comment]
@delip , you are hitting system luarocks rather than ~/torch/install/bin/luarocks for some reason.

Did your install go through succesfully?

**Comment by rosenfeldamir** [comment]
For me, it was a permission problem and the following solution fixed it :+1: 
`sudo chown -R $(whoami) ~/.cache
`
source:
(https://groups.google.com/d/msg/torch7/Vl134tGIU-I/QJfrcNf-CAAJ)

**Comment by msiraj83** [comment]
@Quasimondo ... yes its also work for me and install nngraph and its dependencies from the following links

luarocks --from=https://raw.githubusercontent.com/torch/rocks/master/ install nngraph

**Comment by huhaodong** [comment]
@delip your solution is works for me too,thanks

**Comment by TheGarvitGupta** [comment]
`cd ~/torch/install/bin`
`for NAME in dpnn nn optim optnet csvigo cutorch cunn fblualib torchx tds; do ./luarocks install $NAME; done`

`sudo !!` if required.

**Comment by phongbmatsjsc** [comment]
> luarocks --from=https://raw.githubusercontent.com/torch/rocks/master/ install nngraph

I followed this instruction https://github.com/luarocks/luarocks/wiki/Installation-instructions-for-Windows and successfully installed Lua and LuaRocks. Furthermore, I as well installed dkjson using  "luarocks install dkjson".
But it wasn't going well with nngraph. 
I used this command but the result was
![image](https://user-images.githubusercontent.com/93317158/148518792-ef398039-27b1-4028-8b79-8ec527f4456c.png)
Any advice, guys ?
